src/main/scala/cromwell/binding/WdlNamespace.scala

```diff
@@ -42,345 +128,187 @@ object WdlNamespace {
+
+    val calls = workflowAst.findAsts(AstNodeName.Call) map {Call(_, namespaces, tasks, wdlSyntaxErrorFormatter)}
+    val workflow: Workflow = Workflow(workflowAst, calls)
+
+    // FIXME: This block is run for its side effect of blowing up on the .get (I believe!) - Should there be a real syntax error?
+    // FIXME: It took me a while to understand the logic of the original code & I'm not sure this comment is correct?
+    // FIXME: Also, it'd be nice to move this validity check into Workflow if possible
+    // All MemberAccess ASTs that are not contained in other MemberAccess ASTs
+   for {
+      call <- workflow.calls
+      (name, expression) <- call.inputMappings
+      memberAccess <- expression.ast.findTopLevelMemberAccesses()
+    } yield {
+      getCallFromMemberAccessAst(memberAccess, workflow, wdlSyntaxErrorFormatter).get
     }
 
-    workflowAsts foreach { workflowAst =>
-      val workflow = localWorkflows.head
-      workflowAst.findAsts(AstNodeName.Call) foreach { callAst =>
-        val taskName = callAst.getAttribute("task").sourceString()
-        val taskAst = findTaskAst(taskName) getOrElse {
-          throw new SyntaxError(wdlSyntaxErrorFormatter.callReferencesBadTaskName(callAst, taskName))
-        }
-        val task = findTask(taskName) getOrElse {
-          throw new SyntaxError(wdlSyntaxErrorFormatter.callReferencesBadTaskName(callAst, taskName))
-        }
-
-        val callName = Option(callAst.getAttribute("alias")).getOrElse(callAst.getAttribute("task")).sourceString()
-
-        val call = workflow.calls.find{_.name == callName} getOrElse {
-          throw new SyntaxError(s"Cannot find call with name $callName")
-        }
+    new NamespaceWithWorkflow(namespace, workflow, imports, namespaces, tasks, importedWorkflows, terminalMap, wdlSyntaxErrorFormatter, ast)
+  }
 
-        call.inputMappings foreach { inputKv =>
-          task.inputs find { taskInput => taskInput._1 == inputKv._1 } getOrElse {
-            val callInput = AstTools.callInputAsts(callAst) find {
-              _.getAttribute("key").sourceString == inputKv._1
-            } getOrElse {
-              throw new SyntaxError(s"Can't find call input: ${inputKv._1}")
-            }
-            throw new SyntaxError(wdlSyntaxErrorFormatter.callReferencesBadTaskInput(callInput, taskAst))
-          }
-
-          /* All MemberAccess ASTs that are not contained in other MemberAccess ASTs */
-          inputKv._2.ast.findTopLevelMemberAccesses().map(getCallFromMemberAccessAst).map {_.get}
-        }
-      }
+  /**
+   * Used to safely cast a WdlNamespace to a NamespaceWithWorkflow. Throws an IllegalArgumentException if another
+   * form of WdlNamespace is passed to it
+   */
+  def from(namespace: WdlNamespace): NamespaceWithWorkflow = {
+    namespace match {
+      case n: NamespaceWithWorkflow => n
+      case _ => throw new IllegalArgumentException("Namespace does not have a local workflow to run")
```

Comment:
I suppose I could do an `asInstanceOf` although this exception is cleaner anyways